src/github-handler/branch-handler.ts

```diff
@@ -0,0 +1,102 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Logger, Octokit, RepoDomain} from '../types';
+
+const refPrefix = 'refs/heads/';
+const masterBranch = 'master';
+
+/**
+ * Create a new branch reference with the ref prefix
+ * @param {string} branchName name of the branch
+ */
+function createRef(branchName: string) {
+  return refPrefix + branchName;
+}
+
+/**
+ * Return the length of the GitHub reference prefix
+ */
+function refPrefixLen() {
+  return refPrefix.length;
+}
+
+/**
+ * get primary branch commit HEAD SHA of a repository
+ * Throws an error if the repository cannot be found
+ * or if the primary branch is invalid
+ * @param {Logger} logger The logger instance
+ * @param {Octokit} octokit The authenticated octokit instance
+ * @param {RepoDomain} origin The domain information of the remote origin repository
+ * @param {string} originBranchName The branch name to create on the origin repository
+ * @param {string} primaryBranch the name of the primary branch
+ * @returns {Promise<string>} primary branch commit HEAD SHA
+ */
+async function getPrimaryBranchSHA(
+  logger: Logger,
+  octokit: Octokit,
+  origin: RepoDomain,
+  primaryBranch: string
+): Promise<string> {
+  const branch = (
+    await octokit.repos.getBranch({
+      owner: origin.owner,
+      repo: origin.repo,
+      branch: primaryBranch,
+    })
+  ).data;
+  logger.info('Successfully found primary branch HEAD sha.');
+  return branch.commit.sha;
+}
+
+/**
+ * Create a GitHub branch given a remote origin.
+ * Throws an exception if octokit fails, or if the primary branch is invalid
+ * @param {Logger} logger The logger instance
+ * @param {Octokit} octokit The authenticated octokit instance
+ * @param {RepoDomain} origin The domain information of the remote origin repository
+ * @param {string} originBranchName The branch name to create on the origin repository
+ * @param {string} primaryBranch the name of the primary branch. Default is master
+ * @returns {Promise<string>} the base SHA for subsequent commits to be based off for the origin branch
+ */
+async function branch(
+  logger: Logger,
+  octokit: Octokit,
+  origin: RepoDomain,
+  originBranchName: string,
+  primaryBranch: string = masterBranch
+): Promise<string> {
```

Comment:
Just looking at the argument names, it's a bit hard to tell where we're branch from and to.

If we have the RepoBranch type, we might make this:
```ts
async function branch(
  logger: Logger,
  octokit: Octokit,
  from: RepoBranch,
  to: RepoBranch
): Promise<string>
```

Reply:
There's no from/to relationship. The branch is always created off of the origin with a particular branch name. The primary branch is to get the latest SHA.